src/common/badblock_poolset.c

```diff
@@ -126,3 +126,63 @@ os_badblocks_check_poolset(struct pool_set *set, int create)
 
 	return (cfcb.n_files_bbs > 0);
 }
+
+/*
+ * os_badblocks_clear_poolset_cb -- (internal) callback clearing bad blocks
+ *                                  in the given file
+ */
+static int
+os_badblocks_clear_poolset_cb(struct part_file *pf, void *arg)
+{
+	int *create = arg;
+
+	if (pf->is_remote) { /* XXX not supported yet */
+		ERR(
+			"clearing bad blocks in remote replicas is not supported yet -- '%s'",
+			pf->path);
+		errno = ENOTSUP;
+		return 0;
+	}
+
+	if (*create) {
+		/*
+		 * Poolset is just being created - check if file exists
+		 * and if we can read it.
+		 */
+		int exists = os_access(pf->path, F_OK | R_OK) == 0;
```

Comment:
We should never use access with anything other than F_OK.

From access manpage:
"The check is done using the calling process's real UID and GID, rather than the effective IDs as is done when actually attempting an operation (e.g., open(2)) on the file."
Also I'm not sure whether it works at all, because access manpage says:
"The mode specifies the accessibility check(s) to be performed, and is either the value F_OK, **or** a mask consisting of the bitwise OR of one or more of R_OK, W_OK, and X_OK."